flash/text/embedding/model.py

```diff
@@ -0,0 +1,108 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+import warnings
+from typing import Any, Dict, List, Optional
+
+import torch
+from pytorch_lightning import Callback
+
+from flash.core.integrations.transformers.states import TransformersBackboneState
+from flash.core.model import Task
+from flash.core.registry import FlashRegistry, print_provider_info
+from flash.core.utilities.imports import _TEXT_AVAILABLE
+from flash.core.utilities.providers import _SENTENCE_TRANSFORMERS
+from flash.text.embedding.backbones import HUGGINGFACE_BACKBONES
+from flash.text.ort_callback import ORTCallback
+
+if _TEXT_AVAILABLE:
+    from sentence_transformers.models import Pooling
+
+    Pooling = print_provider_info("Pooling", _SENTENCE_TRANSFORMERS, Pooling)
+
+logger = logging.getLogger(__name__)
+
+
+class SentenceEmbedder(Task):
+    """The ``SentenceEmbedder`` is a :class:`~flash.Task` for generating sentence embeddings, training and
+    validation. For more details, see `embeddings`.
+
+    You can change the backbone to any question answering model from `UKPLab/sentence-transformers
+    <https://github.com/UKPLab/sentence-transformers>`_ using the ``backbone``
+    argument.
+
+    Args:
+        backbone: backbone model to use for the task.
+        enable_ort: Enable Torch ONNX Runtime Optimization: https://onnxruntime.ai/docs/#onnx-runtime-for-training
+    """
+
+    required_extras: str = "text"
+
+    backbones: FlashRegistry = HUGGINGFACE_BACKBONES
+
+    def __init__(
+        self,
+        backbone: str = "sentence-transformers/all-MiniLM-L6-v2",
+        tokenizer_backbone: Optional[str] = None,
+        tokenizer_kwargs: Optional[Dict[str, Any]] = None,
+        enable_ort: bool = False,
+    ):
+        os.environ["TOKENIZERS_PARALLELISM"] = "TRUE"
+        # disable HF thousand warnings
+        warnings.simplefilter("ignore")
+        # set os environ variable for multiprocesses
+        os.environ["PYTHONWARNINGS"] = "ignore"
+        super().__init__()
+
+        if tokenizer_backbone is None:
+            tokenizer_backbone = backbone
+        self.set_state(TransformersBackboneState(tokenizer_backbone, tokenizer_kwargs=tokenizer_kwargs))
+        self.model = self.backbones.get(backbone)()
+        self.pooling = Pooling(self.model.config.hidden_size)
+        self.enable_ort = enable_ort
+
+    def training_step(self, batch: Any, batch_idx: int) -> Any:
+        raise NotImplementedError("Training a `SentenceEmbedder` is not supported. Use a different text task instead.")
+
+    def validation_step(self, batch: Any, batch_idx: int) -> Any:
+        raise NotImplementedError(
+            "Validating a `SentenceEmbedder` is not supported. Use a different text task instead."
+        )
+
+    def test_step(self, batch: Any, batch_idx: int) -> Any:
```

Comment:
What about the predict_step ?

Reply:
We used the forward to generate the embeddings. As for the predict step, it relies on the predict_step implementation in the base task.